tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -895,6 +896,45 @@ static void domain_remove_part_context_switch(void)
 	spawn_user();
 }
 
+void test_stack_buffer_user(void *p1, void *p2, void *p3)
+{
+	volatile char *stack_start = p1;
+	size_t stack_size = (size_t)p2;
+	k_thread_stack_t *stack_obj = p3;
+	u8_t val;
+
+	expect_fault = false;
+	printk("Stack object %p\n", stack_obj);
+	printk("buffer %p\n",
+	       K_THREAD_STACK_BUFFER(stack_obj));
+
+	zassert_equal(stack_start,
+		      K_THREAD_STACK_BUFFER(stack_obj),
+		      "thread->stack_info.start is set wrong");
+
+	/* This should work without crashing */
+	stack_analyze("test_thread", (char *)stack_start, stack_size);
+
+	/* Check that the entire stack buffer is read/writable by user
+	 * mode
+	 */
+	printk("check read/write to stack buffer\n");
+	for (int i = 0; i < stack_size; i++) {
+		val = stack_start[i];
+		stack_start[i] = val;
+	}
+
+	ztest_test_pass();
+}
+
+void test_stack_buffer(void)
+{
+	k_thread_user_mode_enter(test_stack_buffer_user,
+				 (void *)(k_current_get()->stack_info.start),
```

Comment:
This part of the test is incorrect. If dropping to user mode modifies the thread stack info data, the parameters provided will be stale.